src/context.rs

```diff
@@ -741,10 +1033,69 @@ impl DefaultSeafowlContext {
             .ok_or_else(|| {
                 Error::Plan(format!("Schema {schema_name:?} does not exist!"))
             })?;
-        Ok(self
-            .table_catalog
-            .create_table(collection_id, &table_name, &sf_schema)
-            .await?)
+
+        let delta_schema = DeltaSchema::try_from(schema)?;
+
+        // TODO: we could be doing this inside the DB itself (i.e. `... DEFAULT gen_random_uuid()`
+        // in Postgres and `... DEFAULT (uuid())` in SQLite) however we won't be able to do it until
+        // sqlx 0.7 is released (which has libsqlite3-sys > 0.25, with the SQLite version that has
+        // the `uuid()` function).
+        // Then we could create the table in our catalog first and try to create the delta table itself
+        // with the returned uuid (and delete the catalog entry if the object store creation fails).
+        // On the other hand that would complicate etag testing logic.
```

Comment:
It's kind of related to this: https://github.com/splitgraph/seafowl/issues/57 -- the etags should be based on the content hash, rather than on the table version ID, in which case we wouldn't need the UUID to be deterministic.

Reply:
Good point, though I'm also inclined to cover this in a follow-up PR, since it's largely orthogonal to the delta-rs migration.